src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -98,37 +111,54 @@ synchronized void createTable(String tableName, List<Field<?>> columns, String r
                                 String referenceTablePrimaryKeyFieldName) throws SQLException {
     Field foreignKeyField = DSL.field(referenceTablePrimaryKeyFieldName, Integer.class);
     columns.add(foreignKeyField);
-    Table referenceTable = DSL.table(referenceTableName);
-    CreateTableConstraintStep constraintStep = create.createTable(tableName)
-        .column(DSL.field(getPrimaryKeyColumnName(tableName) + PRIMARY_KEY_AUTOINCREMENT_POSTFIX))
-        .columns(columns)
-        .constraints(DSL.constraint(foreignKeyField.getName() + "_FK").foreignKey(foreignKeyField)
-            .references(referenceTable, DSL.field(referenceTablePrimaryKeyFieldName)));
 
-    LOG.debug("table with fk created: {}", constraintStep.toString());
     try {
+      LOG.debug("Trying to create a summary table: {} that references {}", tableName, referenceTableName);
+      Table referenceTable = DSL.table(referenceTableName);
+      CreateTableConstraintStep constraintStep = create.createTable(tableName)
+          .column(DSL.field(getPrimaryKeyColumnName(tableName) + PRIMARY_KEY_AUTOINCREMENT_POSTFIX))
+          .columns(columns)
+          .constraints(DSL.constraint(foreignKeyField.getName() + "_FK").foreignKey(foreignKeyField)
+              .references(referenceTable, DSL.field(referenceTablePrimaryKeyFieldName)));
       constraintStep.execute();
-      jooqTableColumns.put(tableName, columns);
-    } catch (Exception e) {
-      LOG.error("Failed to create table {}", tableName);
-      throw new SQLException();
+      LOG.debug("table with fk created: {}", constraintStep.toString());
+    } catch (DataAccessException e) {
+      String msg = "table " + tableName + " already exists";
+      if (e.getMessage().contains(msg)) {
+        LOG.debug(e.getMessage());
+      } else {
+        LOG.error("Error creating table: {}", tableName, e);
+        throw new SQLException(e);
+      }
     }
+    tableNames.add(tableName);
+    jooqTableColumns.put(tableName, columns);
   }
 
   @Override
   synchronized int insertRow(String tableName, List<Object> row) throws SQLException {
     int lastPrimaryKey = -1;
     String sqlQuery = "SELECT " + LAST_INSERT_ROWID;
-    InsertValuesStepN insertValuesStepN = create.insertInto(DSL.table(tableName))
-        .columns(jooqTableColumns.get(tableName))
-        .values(row);
-    LOG.debug("sql insert: {}", insertValuesStepN.toString());
+
+    Objects.requireNonNull(create, "DSLContext cannot be null");
+    Table<Record> table = DSL.table(tableName);
+    List<Field<?>> columnsForTable = jooqTableColumns.get(tableName);
+    if (columnsForTable == null) {
+      LOG.error("NO columns found for table: {}. Tables: {}, columns: {}", tableName, tableNames, jooqTableColumns);
+      throw new SQLException("No columns exist for table.");
+    }
+
     try {
+      InsertValuesStepN insertValuesStepN = create
```

Comment:
same as above

Reply:
This create statement is less involved and therefore changes of exceptions are less. Will move it out.